packages/sdk/src/util/webHost.ts

```diff
@@ -67,11 +75,8 @@ export class WebHost {
 			// host static binaries
 			(req, res, next) => this.serveStaticBuffers(req, res, next),
 			// host static files
-			Restify.plugins.serveStatic({
-				directory: this._baseDir,
-				default: 'index.html'
-			}
-			));
+			Restify.plugins.serveStaticFiles(this._baseDir, { index: 'index.html', etag: true }),
+			Restify.plugins.conditionalRequest());
```

Comment:
I think we want to use `Cache-Control: no-cache` semantics, meaning that every cache hit on the client would result in an etag check and possible re-download (by sending the `If-None-Match` header with the last known etag). There would be no set `max-age` on the resource. Will the client side need to be updated to send the `If-None-Match` header, and handle a 304/Not Modified result code?

Reply:
Yes, the client today cannot take advantage of any of this code. I'm working on a design doc for an update client asset cache system, and this will be a dependency of almost any system I cook up.

Reply:
I imagine modifying the client asset cache to store etag and last-modified alongside the processed object. Then when you need the object. attempt the HTTP request first with those headers, then fall back to the cached version on 304.

Reply:
Here's a nice overview of control with with etags: https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/http-caching